pds4indextools/pds4_create_xml_index.py

```diff
@@ -434,26 +532,50 @@ def main():
         namespaces['pds'] = namespaces.pop(None)
         prefixes = {v: k for k, v in namespaces.items()}
 
-        xml_results = {}
-        traverse_and_store(root, tree, xml_results, elements_to_scrape,
+        label_results = {}
+        traverse_and_store(root, tree, label_results, elements_to_scrape,
                            nillable_elements_info, config, file)
 
-        for key in list(xml_results.keys()):
-            process_tags(xml_results, key, root,
-                         namespaces, prefixes, args)
+        for key in list(label_results.keys()):
+            process_tags(label_results, key, root,
+                         namespaces, prefixes)
+            
+        new_xpaths = renumber_xpaths(label_results.keys())
+        for key, value in new_xpaths.items():
+            label_results[value] = label_results.pop(key)
+
+        if args.disambiguate_xpaths:
+                elements = ()
+                xpath_elements = []
+                tags = []
+                for key in list(label_results.keys()):
+                    xpath_elements.append(grab_elements(key))
+                
+                duplicates = [t for t in set(xpath_elements) if xpath_elements.count(t) > 1]
+
+                for key in list(label_results.keys()):
```

Comment:
So here is one case where you actually DO need to use list, because you're modifying label_results inside the loop. If you don't use list() to create a copy of the keys first, you will end up modifying the contents of the dictionary while you're iterating over it, which will give bad results (and usually even a Python warning to let you know you did it).